src/main/java/com/example/SpringCrud/service/UserService.java

```diff
@@ -51,17 +51,20 @@ public User deleteUser(Long id) {
 
     @Override
     public User borrowBook(Long userId, Long bookId) {
-        User theBorrower = userRepository.findById(userId).orElse(null);
-        Book toBorrow = bookRepository.findById(bookId).orElse(null);
-        theBorrower.setBorrowedBooks(toBorrow);
-        return userRepository.save(theBorrower);
+          User theBorrower = userRepository.findById(userId).orElse(null);
+          Book toBorrow = bookRepository.findById(bookId).orElse(null);
+          theBorrower.setBorrowedBooks(toBorrow);
+          toBorrow.setUser(theBorrower);
+          theBorrower.setBorrowedBooks(toBorrow);
+          return userRepository.save(theBorrower);
     }
 
     @Override
     public User returnBook(Long userId, Long bookId) {
         User theBorrower = userRepository.findById(userId).orElse(null);
         Book toReturn = bookRepository.findById(bookId).orElse(null);
         theBorrower.returnBook(toReturn);
```

Comment:
Do a null check here to verify the user of book exists in the db to avoid the following error:-
java.lang.NullPointerException\n\tat com.example.SpringCrud.controller.UserController.returnBook